### ISSUE TITLE
React, MFE - Timed out waiting for remote to start. Check above for any errors

### ISSUE THREAD

**Issue by vadimka123** [nrwl/nx#17354]
### Current Behavior

After some time after successful start I get this error
Started after update to 16.3.0

### Expected Behavior

All continue work successfully 

### Steps to Reproduce

I checked code and looks like related to custom host option instead of standard localhost 

nx serve host --open --devRemotes={...} --ssl --host {...} --ssl-cert {...}--ssl-key {...}

### Nx Report

```shell
Node   : 18.12.1
   OS     : darwin x64
   npm    : 8.19.2
   Hasher : Native
   
   nx                 : 16.3.0
   @nx/js             : 16.3.0
   @nx/jest           : 16.3.0
   @nx/linter         : 16.3.0
   @nx/workspace      : 16.3.0
   @nx/cypress        : 16.3.0
   @nx/devkit         : 16.3.0
   @nx/eslint-plugin  : 16.3.0
   @nx/react          : 16.3.0
   @nrwl/tao          : 16.3.0
   @nx/web            : 16.3.0
   @nx/webpack        : 16.3.0
   nx-cloud           : 16.0.5
   typescript         : 5.0.4
```

### Failure Logs

```shell
.../node_modules/@nx/react/src/executors/module-federation-dev-server/module-federation-dev-server.impl.js:98
                throw new Error(`Timed out waiting for remote to start. Check above for any errors.`);
                      ^

Error: Timed out waiting for remote to start. Check above for any errors.
    at .../node_modules/@nx/react/src/executors/module-federation-dev-server/module-federation-dev-server.impl.js:98:23
    at Generator.throw (<anonymous>)
    at rejected (.../node_modules/tslib/tslib.js:165:69)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Node.js v18.12.1

 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Running target serve for project host failed

   Failed tasks:
   
   - host:serve:development
   
   Hint: run the command with --verbose for more details.

   View structured, searchable error logs at ...
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by vadimka123** [comment]
@jaysoo , can you please look in to this issue?
Looks like need to change logic - check hostname from options if they provided otherwise only localhost?

**Comment by Ifunanyacollins** [comment]
@xiongemi, any update to fix this issue?

**Comment by senseibarni** [comment]
I have the same issue after upgrading to 16.5.0 (from 16.2.2).

**Comment by WizzApp** [comment]
I'm sorry to chime in here, but the issue is still happening with 16.5.5. Is this fixed in 16.6?  I just saw that the PR was closed. Is there any workaround for this problem?

**Comment by morkuj** [comment]
The issue still occurs if serving remotes using the custom host option instead of the default "localhost".
